recipes/protobuf/all/conanfile.py

```diff
@@ -63,18 +65,35 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
+        if Version(self.version) >= "3.22.0":
+            self.requires("abseil/20230125.3", transitive_headers=True)
         if self.options.with_zlib:
             self.requires("zlib/[>=1.2.11 <2]")
 
     def validate(self):
         if self.options.shared and is_msvc_static_runtime(self):
             raise ConanInvalidConfiguration("Protobuf can't be built with shared + MT(d) runtimes")
-
-        check_min_vs(self, "190")
-
-        if self.settings.compiler == "clang":
-            if Version(self.settings.compiler.version) < "4":
-                raise ConanInvalidConfiguration(f"{self.ref} doesn't support clang < 4")
+        if Version(self.version) < "3.22.0":
+            check_min_vs(self, "190")
+            if self.settings.compiler == "clang":
+                if Version(self.settings.compiler.version) < "4":
+                    raise ConanInvalidConfiguration(f"{self.ref} doesn't support clang < 4")
+        else:
+            if self.settings.compiler.cppstd:
+                check_min_cppstd(self, self._min_cppstd)
+            else:
+                raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._min_cppstd}, cppstd must be specified.")
```

Comment:
The cppstd is checked at compile time, so I don't have a better solution if it is not defined by default.

Reply:
Would it not be defined by the CMake build system to be 11 or 14 if it needs to? I don't expect not defining it to break anything. A ton of other CCI recipes use this pattern, but there's no `else` there. Maybe I'm misunderstanding the problem though?

Reply:
check_min_cppstd on its own without the "if check" seems to be working fine.

Reply:
Still looks like all of the v1 checks are invalid. There's a line in the implementation that checks for cppstd to be defined:
```python
    current_cppstd = conanfile.settings.get_safe("compiler.cppstd")
    if current_cppstd is None:
        raise ConanInvalidConfiguration("The compiler.cppstd is not defined for this configuration")
```
So I think that check still needs to be guarded, and the v1 builds fixed.

Reply:
The doc for check_min_cppstd says: "Validates if the applied cppstd setting (from compiler.cppstd settings or deducing the default from compiler and compiler.version) is at least the value specified in the cppstd argument. "

The v1 builds have been run already ?

Reply:
The check only happens if the cppstd is defined, and the implementation suggests that if it isn't, the package is automatically invalid.

The v1 checks pass, but because nothing ends up being built.

Reply:
The v1 build was not skipped : https://c3i.jfrog.io/c3i/misc/logs/pr/23095/39-linux-gcc-legacy/protobuf/3.21.12//7627fae1426bcc12a67dba7c7207b1bccf05e5fd-build.txt

Reply:
Specifically the checks for the new version were skipped, the others are run because of the version check above: https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/23095/39/protobuf/4.25.0//summary.json

Reply:
Sorry about any confusion!

I think the build systems for each should correctly set the C++ standard, it's just that Conan (in the v1 pipeline) doesn't know what it is, so it can't check if it's valid. I believe that's the reason it's recommended in Conan v2. So I think just have the check be guarded by `if self.settings.compiler.get_safe("cppstd"):` should be sufficient.

Reply:
One of the compiler failed for the check `#if __cplusplus < 201402L` at build time.